akka-http-caching/src/main/scala/akka/http/routing/directives/CachingDirectives.scala

```diff
@@ -0,0 +1,92 @@
+package akka.http.routing.directives
+
+import akka.actor.ActorRefFactory
+import akka.http.caching.{ Cache, LfuCache }
+import akka.http.scaladsl.server.Directive0
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ ExecutionContext, Future }
+import akka.http.scaladsl.server._
+import akka.http.scaladsl.model.HttpMethods.GET
+import akka.http.scaladsl.model.headers._
+import akka.http.scaladsl.model.headers.CacheDirectives._
+import akka.http.scaladsl.server.RouteResult.Rejected
+
+trait CachingDirectives {
```

Comment:
Each exposed directives should be documented in a separate page in the `docs` project and listed in the [alphabetically](https://github.com/akka/akka-http/blob/0106fc93a05debd9276eafcf2dde94faee4cba15/docs/src/main/paradox/scala/http/routing-dsl/directives/alphabetically.md) and [by-trait](https://github.com/akka/akka-http/blob/0106fc93a05debd9276eafcf2dde94faee4cba15/docs/src/main/paradox/scala/http/routing-dsl/directives/by-trait.md) pages. The [spray cache directives docs](https://github.com/spray/spray/tree/master/docs/documentation/spray-routing/caching-directives) look like a good start.

Some minimal documentation introducing caching based on
https://github.com/spray/spray/blob/master/docs/documentation/spray-caching/index.rst would also be great. This would also be a good place to make it clear that using the caching directives requires adding an additional dependency.